specification/query_ruleset/_types/QueryRuleset.ts

```diff
@@ -0,0 +1,78 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { IndexName, Name } from '@_types/common'
+import { EpochTime, UnitMillis } from '@_types/Time'
+import { InlineScript } from '@_types/Scripting'
+import { UserDefinedValue } from '@spec_utils/UserDefinedValue'
+
+export class QueryRuleset {
+  /**
+   * Query Ruleset unique identifier
+   */
+  ruleset_id: QueryRulesetId
+  /**
+   * Rules associated with the query ruleset
+   */
+  rules: QueryRule[]
+}
+
+export type QueryRulesetId = string
```

Comment:
Maybe using `Id` from `_types/common` would make more sense than defining a new type (even though it still maps to `string`) 😉

Reply:
That's a good callout, thanks.